### PR TITLE
Implementing styles for checkbox and radio button

### PR DESCRIPTION
**Pull request by donaldkibet**
Screenshot from 2019-12-04 16-55-56

### CODE REVIEW

File: src/buttons-and-links/buttons-and-links.css [hunk 1]
```diff
@@ -58,7 +58,6 @@ a.omrs-unstyled:active:hover {
   border: inherit;
   outline: inherit;
   box-shadow: inherit;
-  padding: inherit;
```
Comment (joeldenning) [inline src/buttons-and-links/buttons-and-links.css]: Why is this removed?

File: src/forms/forms.stories.html [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  .stories-example > input {
+    margin: 35px;
+  }
+
+  p.type-demo {
+    text-transform: capitalize;
+  }
+
+  .omrs-input-group {
```
Comment (joeldenning) [inline src/forms/forms.stories.html]: Since this css class isn't exported from the styleguide, I don't think it should be named with the `omrs-` prefix. People looking at the HTML for inputs might be confused when they see this. It might be best to do a selector like this so that the copy/pasteable example code doesn't have css classes that aren't in the styleguide:

```css
.form-examples form > div {
}
```

```html
<article class="form-examples">
</article>
```

^ this avoids the code in the examples not referencing css classes that aren't actually part of the styleguide.

File: src/forms/forms.css [hunk 3]
```diff
@@ -0,0 +1,114 @@
+.omrs-checkbox {
+  display: inline-block;
+  vertical-align: middle;
+  user-select: none;
+  cursor: pointer;
+}
+
+.omrs-checkbox input[type="checkbox"] {
+  opacity: 0;
+  position: absolute;
```
Comment (joeldenning) [inline src/forms/forms.css]: I don't think we should set this as absolutely positioned, if we can avoid it. The reason is that absolutely positioned elements require a [positioned ancestor](https://stackoverflow.com/questions/45784777/what-is-a-positioned-ancestor). We should not require the user of the styleguide to provide a positioned ancestor for all inputs.

If absolute positioning is required for the checkbox to look correct, we should consider making the `<div class="omrs-checkbox">` element `position: relative` so that it becomes the positioned ancestor for the `<input>`

File: src/forms/forms.css [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+.omrs-checkbox label {
+  position: relative;
+  padding-left: 1.5rem;
+}
+
+.omrs-checkbox label::before,
+.omrs-checkbox label::after {
+  position: absolute;
```
Comment (joeldenning) [inline src/forms/forms.css]: Why is this needed? What is its positioned ancestor?

File: src/forms/forms.css [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+.omrs-radio-button {
+  display: inline-block;
+  vertical-align: middle;
+  user-select: none;
+  cursor: pointer;
+}
+
+.omrs-radio-button input[type="radio"] {
+  opacity: 0;
+  position: absolute;
```
Comment (joeldenning) [inline src/forms/forms.css]: Same here about positioning elements without knowing what the positioned ancestor will be

File: src/forms/forms.stories.html [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    <div class="examples">
+      <div class="tabs">
+        <div lang="html" class="tab">HTML</div>
+        <div lang="jsx" class="tab">JSX</div>
+      </div>
+      <div class="panels">
+        <div class="panel html">
+          <div class="stories-example lang-html">
+            <div className="omrs-checkbox">
+              <input type="checkbox" id="inputHtmlCheckBox" />
```
Comment (joeldenning) [inline src/forms/forms.stories.html]: Inputs usually should have `name` attributes. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#name. Would be good to include the name in the documentation.

File: src/forms/forms.stories.html [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+              <label for="inputHtmlCheckBox">CheckBox HTML</label>
+            </div>
+          </div>
+          <div class="stories-example lang-html">
+            <div className="omrs-radio-button">
+              <input type="radio" id="inputHtmlRadio" />
```
Comment (joeldenning) [inline src/forms/forms.stories.html]: Same here about a `name` attribute

File: src/forms/forms.stories.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      True: "checked",
+      False: ""
+    },
+    "checked"
+  );
+
+  const determineInputType = inputType =>
+    inputType === "checkbox" ? "checkbox" : "radio";
+
+  const determinStyle = inputType =>
```
Comment (joeldenning) [inline src/forms/forms.stories.js]:
```suggestion
  const determineStyle = inputType =>
```

File: src/buttons-and-links/buttons-and-links.css (hunk 1 above)
Comment (donaldkibet) [reply src/buttons-and-links/buttons-and-links.css]: placed this back

**Comment by donaldkibet** [comment]
> The `src/forms/index.html` file seems like it could be deleted entirely, right?
> 
> Nice work on this. One question - how hard would it be to make it possible to support the following html contract?
> 
> ```
> <div class="omrs-checkbox">
>   <label>
>     <input type="checkbox" name="thing" />
>     The label text
>   </label>
> </div>
> ```
> 
> ^ this is nice because it doesn't require html `id` attributes. However, sometimes the `id` attribute is also nice, so it would be cool to support both. Might be hard to support that though. What do you think?

Not difficult i have implemented it as recommended

File: src/forms/forms.stories.html (hunk 7 above)
Comment (donaldkibet) [reply src/forms/forms.stories.html]: Included the `name` attributed

File: src/forms/forms.css (hunk 4 above)
Comment (donaldkibet) [reply src/forms/forms.css]: required this to position the `::after and ::before` content at the same position to implement the custom checkbox mark

File: src/forms/forms.css (hunk 3 above)
Comment (donaldkibet) [reply src/forms/forms.css]: resolved this `.omrs-checkbox label` is the ancestor element

**Comment by joeldenning** [comment]
What I said is that it would be good to have it work **both ways** (both with input inside of label and with input as sibling of label). It might not be worth implementing that, though.

**Comment by jonathandick** [comment]
@donaldkibet , any updates on last comment from @joeldenning ?

**Comment by donaldkibet** [comment]
> @donaldkibet , any updates on last comment from @joeldenning ?

working on it today

**Comment by joeldenning** [comment]
I am no longer going to review this. @fatmali @brandones can you take over

File: src/forms/forms.stories.html [hunk 9]
```diff
@@ -0,0 +1,193 @@
+<article>
+  <h2 class="title">OpenMRS Forms Elements</h2>
+  <p class="title">
+    There are 2 types of form elements defined in the styleguide
```
Comment (fatmali) [inline src/forms/forms.stories.html]: 2 or 3? I see you have `datepicker` styles defined as well
